consensus/src/rand/rand_gen/rand_manager.rs

```diff
@@ -48,6 +48,94 @@ use tokio_retry::strategy::ExponentialBackoff;
 pub type Sender<T> = UnboundedSender<T>;
 pub type Receiver<T> = UnboundedReceiver<T>;
 
+pub(crate) struct ExtensionV1<D: TAugmentedData> {
+    aug_data_store: AugDataStore<D>,
+}
+
+impl<D: TAugmentedData> ExtensionV1<D> {
+    pub(crate) fn new(
+        epoch_state: Arc<EpochState>,
+        signer: Arc<ValidatorSigner>,
+        config: RandConfig,
+        fast_config: Option<RandConfig>,
+        db: Arc<dyn RandStorage<D>>,
+    ) -> Self {
+        let aug_data_store = AugDataStore::new(
+            epoch_state.epoch,
+            signer,
+            config.clone(),
+            fast_config.clone(),
+            db,
+        );
+        Self { aug_data_store }
+    }
+}
+
+/// In v2, apk is broadcasted forever.
```

Comment:
How is broadcasting forever implemented? the ReliableBroadcast will stop when receiving the ack from quorum?

Reply:
the current implementation does not respond to the rb request.

Reply:
I see. Then instead of broadcasting forever, why not fetch missing APK when verifying shares? Otherwise the restarted node will need to wait for 10s if unlucky.

Reply:
i'm not sure which way to go.
if fetching on demand, do we also need some rate limiting logic?

Reply:
Let's see what @zekun000 thinks

Reply:
I don't like the broadcasting forever too, broadcast + fetching on miss seems reasonable, not sure if it's difficult to implement. I'm not too worried about rate limiting, it should be simple to add if necessary